server/server.go

```diff
@@ -108,11 +108,15 @@ func New(opts ...Option) *Server {
 	for _, opt := range opts {
 		opt(cfg)
 	}
+	url := ""
 	if len(cfg.endpoints) == 0 {
-		log.Fatalf("No endpoints defined!")
+		log.Printf("No endpoints defined!")
```

Comment:
I'd remove the log statement here. Either we validate in the `New` function and return an error or we check later. If we don't want to change the signature of `New` then we check later and you have a check for that.